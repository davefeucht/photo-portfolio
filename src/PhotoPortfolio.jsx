/*******************
* Top level PhotoPortfolio component. Sets up the base site URL and 
* loads the title bar and list of categories.
*******************/

import React from "react";
import TitleBar from "./TitleBar.jsx";
import Categories from "./Categories.jsx";
import Footer from "./Footer.jsx";

export default class PhotoPortfolio extends React.Component {
  constructor(props) {
    super(props);

    //Set up the base site url
    this.site_url = "throughapinhole.com";

  }

  render () {

    return (
      <div className="app">
        <TitleBar />
        <div className="content">
          <Categories site={this.site_url} />
        </div>
        <Footer />
      </div>
    );
  }
}