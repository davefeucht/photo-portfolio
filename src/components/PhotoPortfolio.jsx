/*******************
* Top level PhotoPortfolio component. Sets up the base site URL and 
* loads the title bar and list of categories.
*******************/

import React from "react";
import axios from "axios";
import App from "./styledComponents/App.jsx";
import TitleBar from "./TitleBar.jsx";
import MainContent from "./styledComponents/MainContent.jsx";
import Categories from "./Categories.jsx";
import Category from "./Category.jsx";
import Footer from "./Footer.jsx";

export default class PhotoPortfolio extends React.Component {

  static site_url = "throughapinhole.com";

  state = {
    siteName: "",
    showAllCategories: true,
    singleCategoryToShow: {id: 0, name: ""},
  }

  _getSiteInformation() {
    const getSiteInformationURI = `https://${PhotoPortfolio.site_url}/wp-json/`;

    axios.get(getSiteInformationURI)
      .then(response => {
        const siteName = response.data.name;
        this.setState({siteName});
      })
      .catch(error => {
        console.log(error.message);
      })
  }

  //Method to display a single category when one is clicked on.
  _showSpecificCategory(categoryId, categoryName) {
    this.setState({showAllCategories: false}); 
    this.setState({singleCategoryToShow: {id: categoryId, name: categoryName}}); 
  }
  
  //Method to display all existing categories.
  _showAllCategories() {
    this.setState({showAllCategories: true});
  }

  componentWillMount() {
    this._getSiteInformation();
  }

  render () {
    let contentToDisplay = <Categories site={PhotoPortfolio.site_url} showSingleCategory={this._showSpecificCategory.bind(this)} />

    if(!this.state.showAllCategories) {
      contentToDisplay = <Category key={this.state.singleCategoryToShow.id.toString()} id={this.state.singleCategoryToShow.id} name={this.state.singleCategoryToShow.name} site={PhotoPortfolio.site_url} clickCategory={this._showAllCategories.bind(this)} />; 
    }

    return (

      <App>
        <TitleBar siteName={this.state.siteName} />
        <MainContent>
          {contentToDisplay}
        </MainContent>
        <Footer />
      </App>
    );
  }
}