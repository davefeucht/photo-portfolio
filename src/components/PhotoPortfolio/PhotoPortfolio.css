.photo-portfolio {
  grid-area: content;
}

.app {
  height: 100vh;
  display: grid;
  grid-template-areas: "header" "content" "footer";
  grid-template-rows: 3rem auto 2rem;
}