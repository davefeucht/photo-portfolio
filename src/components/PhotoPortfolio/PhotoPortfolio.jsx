/** *****************
* Top level PhotoPortfolio component. Sets up the base site URL and 
* loads the title bar and list of categories.
****************** */

import './PhotoPortfolio.css';

import { reaction, runInAction } from 'mobx';
import { observer } from 'mobx-react';
import PropTypes from 'prop-types';
import React, { useEffect } from 'react';
import {
    Route,
    Routes,
    useParams
} from 'react-router-dom';

import { getCategories, getPages } from '../../utils/Api';
import Categories from '../Categories/Categories.jsx';
import Category from '../Category/Category.jsx';
import Footer from '../Footer/Footer.jsx';
import Menu from '../Menu/Menu.jsx';
import Page from '../Page/Page.jsx';
import Post from '../Post/Post.jsx';
import TitleBar from '../TitleBar/TitleBar.jsx';

function PhotoPortfolio({ stateStore }) {
    const { categoryId, pageId } = useParams();

    const setScreenSize = () => {
        runInAction(() => {
            stateStore.screenInfo.width = stateStore.applicationRoot.clientWidth;
            stateStore.screenInfo.height = stateStore.applicationRoot.clientHeight;
        });
    };

    const setColumns = (width, height) => {
        if ((width < height)) {
            document.body.style.setProperty('--number-of-columns', 2);
        } else {
            document.body.style.setProperty('--number-of-columns', 5);
        }
    };

    useEffect(() => {
        window.addEventListener('resize', setScreenSize.bind(this));
        getCategories(stateStore.siteInfo.siteUrl)
            .then(categories => {
                stateStore.setCategoryList(categories);
            });
        getPages(stateStore.siteInfo.siteUrl)
            .then(pages => {
                stateStore.setPages(pages);
            });

        setScreenSize();
        setColumns(stateStore.screenInfo.width, stateStore.screenInfo.height);
        const disposer = reaction(
            () => [stateStore.applicationRoot.clientWidth, stateStore.applicationRoot.clientHeight],
            () => setScreenSize(stateStore.applicationRoot.clientWidth, stateStore.applicationRoot.clientHeight)
        );
        const disposer2 = reaction(
            () => [stateStore.screenInfo.width, stateStore.screenInfo.height],
            () => setColumns(stateStore.screenInfo.width, stateStore.screenInfo.height)
        );
        return () => {
            window.removeEventListener('resize', setScreenSize);
            disposer();
            disposer2();
        };
    }, []);

    return (
        <div className="app">
            <TitleBar stateStore={stateStore} />
            <div className="photo-portfolio">
                <Routes>
                    <Route
                        path="/"
                        element={<Categories stateStore={stateStore} />}
                    />
                    <Route
                        path="page/:pageId"
                        element={<Page key={pageId} stateStore={stateStore} />}
                    />
                    <Route
                        path="category/:categoryId"
                        element={<Category key={categoryId} stateStore={stateStore} />}
                    >
                        <Route
                            path="post/:postId"
                            element={<Post stateStore={stateStore} />}
                        />
                    </Route>
                </Routes>
            </div>
            <Menu stateStore={stateStore} />
            <Footer />
        </div>
    );
}

PhotoPortfolio.displayName = 'PhotoPortfolio';

PhotoPortfolio.propTypes = {
    stateStore: PropTypes.object.isRequired
};

export default observer(PhotoPortfolio);
