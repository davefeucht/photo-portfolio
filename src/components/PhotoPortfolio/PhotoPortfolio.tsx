/** *****************
* Top level PhotoPortfolio component. Sets up the base site URL and 
* loads the title bar and list of categories.
****************** */

import './PhotoPortfolio.css';

import { reaction, runInAction } from 'mobx';
import { observer } from 'mobx-react';
import * as React from 'react';
import { useEffect } from 'react';
import {
    Route,
    Routes,
    useParams
} from 'react-router-dom';

import { getCategories, getPages } from '../../utils/Api';
import { Store } from '../../utils/types';
import Categories from '../Categories/Categories';
import Category from '../Category/Category';
import Footer from '../Footer/Footer';
import Menu from '../Menu/Menu';
import Page from '../Page/Page';
import Post from '../Post/Post';
import TitleBar from '../TitleBar/TitleBar';

interface PhotoPortfolioProps {
    stateStore: Store
}

const PhotoPortfolio: React.FC<PhotoPortfolioProps> = ({ stateStore }) => {
    const { categoryId, pageId } = useParams();

    const setScreenSize = () => {
        runInAction(() => {
            stateStore.screenInfo.width = stateStore.applicationRoot.clientWidth;
            stateStore.screenInfo.height = stateStore.applicationRoot.clientHeight;
        });
    };

    const setColumns = (width: number, height: number) => {
        if ((width < height)) {
            document.body.style.setProperty('--number-of-columns', '2');
        } else {
            document.body.style.setProperty('--number-of-columns', '5');
        }
    };

    useEffect(() => {
        window.addEventListener('resize', setScreenSize.bind(this));
        getCategories(stateStore.siteInfo.siteUrl)
            .then(categories => {
                stateStore.setCategoryList(categories);
            });
        getPages(stateStore.siteInfo.siteUrl)
            .then(pages => {
                stateStore.setPages(pages);
            });

        setScreenSize();
        setColumns(stateStore.screenInfo.width, stateStore.screenInfo.height);
        const disposer = reaction(
            () => [stateStore.applicationRoot.clientWidth, stateStore.applicationRoot.clientHeight],
            () => setScreenSize()
        );
        const disposer2 = reaction(
            () => [stateStore.screenInfo.width, stateStore.screenInfo.height],
            () => setColumns(stateStore.screenInfo.width, stateStore.screenInfo.height)
        );
        return () => {
            window.removeEventListener('resize', setScreenSize);
            disposer();
            disposer2();
        };
    }, []);

    return (
        <div className="app">
            <TitleBar stateStore={stateStore} />
            <div className="photo-portfolio">
                <Routes>
                    <Route
                        path="/"
                        element={<Categories stateStore={stateStore} />}
                    />
                    <Route
                        path="page/:pageId"
                        element={<Page key={pageId} stateStore={stateStore} />}
                    />
                    <Route
                        path="category/:categoryId"
                        element={<Category key={categoryId} stateStore={stateStore} />}
                    >
                        <Route
                            path="post/:postId"
                            element={<Post stateStore={stateStore} />}
                        />
                    </Route>
                </Routes>
            </div>
            <Menu stateStore={stateStore} />
            <Footer />
        </div>
    );
};

PhotoPortfolio.displayName = 'PhotoPortfolio';

export default observer(PhotoPortfolio);
