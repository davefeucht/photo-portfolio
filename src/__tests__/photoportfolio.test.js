import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import PhotoPortfolio from '../components/PhotoPortfolio/PhotoPortfolio';
import stateStore from '../StateStore/store';

const pages = [
    {
        id: 150,
        content: {
            rendered: 'Contact me!'
        },
        title: {
            rendered: 'Contact'
        }
    }
]

const categories = [
    {
        id: 23,
        count: 10,
        description: 'Category of other things',
        link: '',
        name: 'Some other category',
        slug: 'smocg',
        taxonomy: '',
        parent: 12,
        meta: {}
    },
    {
        id: 35,
        count: 3,
        description: 'Category of things',
        link: '',
        name: 'Some category',
        slug: 'smcg',
        taxonomy: '',
        parent: 23,
        meta: {}
    }
];

jest.mock('../utils/Api', () => ({
    getCategories: () => Promise.resolve(categories),
    getPages: () => Promise.resolve(pages)
}));

test('PhotoPortfolio displays', () => {
  const store = new stateStore();
  let component;
  let tree;
  component = renderer.create(
      <MemoryRouter initialEntries={['/category/35']} >
          <PhotoPortfolio stateStore={store}></PhotoPortfolio>
      </MemoryRouter>
  );
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
