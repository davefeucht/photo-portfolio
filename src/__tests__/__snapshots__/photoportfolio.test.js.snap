// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoPortfolio displays 1`] = `
<div
  className="app"
>
  <div
    className="title-bar"
  >
    <div
      className="hamburger-button closed"
      onClick={[Function]}
    >
      <img
        border="0"
        src="./assets/images/hamburger_icon.svg"
      />
    </div>
    <div>
      <h1 />
    </div>
  </div>
  <div
    className="photo-portfolio"
  >
    <div
      className="category"
    >
      <div
        className="section-header"
      >
        <div
          className="title"
        />
      </div>
      <div
        className="posts"
      />
    </div>
  </div>
  <div
    className="menu closed"
  >
    <ul>
      <li
        onClick={[Function]}
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <div
            className="menu-link"
          >
            Home
          </div>
        </a>
      </li>
      <li>
        Categories
        <ul />
      </li>
      <li>
        Info
        <ul />
      </li>
    </ul>
  </div>
  <div
    className="footer"
  >
    Theme Copyright @2021 Dave Feucht
  </div>
</div>
`;
